pkg/receive/handler.go

```diff
@@ -964,20 +967,34 @@ func (h *Handler) replicate(ctx context.Context, tenant string, wreq *prompb.Wri
 		return errors.New("hashring is not ready")
 	}
 
-	for i = 0; i < h.options.ReplicationFactor; i++ {
-		endpoint, err := h.hashring.GetN(tenant, &wreq.Timeseries[0], i)
-		if err != nil {
-			h.mtx.RUnlock()
-			return err
+	replicatedRequests := make(map[endpointReplica]*prompb.WriteRequest)
+	for i := uint64(0); i < h.options.ReplicationFactor; i++ {
+		for _, ts := range wreq.Timeseries {
```

Comment:
It's not sufficient to rehash the first series in the batch anymore. This works for `hashmod`, but cannot be generally applied to other algorithms.